### PR TITLE
Added pinecone support for the ChatQnA application

### PR DESCRIPTION
**Pull request by pallavijaini0525**
## Type of Change

Added pinecone support for the ChatQnA application

## Description

I updated the server.py class in the qna-app/app directory to initialize the vector store according to the vectordb argument. Additionally, I made corresponding changes in the utils.py helper class to handle this functionalit,y

I included the "pinecone" folder containing the "ingest.py" and rag_pinecone folder with "chain.py" files and more.

## How has this PR been tested?

This pull request has been tested, following the same procedure as the existing RAG flow. The main difference is that the vectordb parameter was set to "pinecone", and a few additional environment variables were configured in the docker-compose-pinecone.yaml file to use the pinecone vectordb. Did the regression testing for redis-db as well.

how to reproduce the test (including hardware information)

## Dependency Change?

langchain-pinecone

### CODE REVIEW

**Comment by lvliang-intel** [comment]
@pallavijaini0525 
Please handle the pre-CI issues.

ChatQnA/langchain/docker/qna-app/app/server.py:140:14: F821 Undefined name `vectordb`
ChatQnA/langchain/docker/qna-app/app/utils.py:307:23: F821 Undefined name `Pinecone`
ChatQnA/langchain/docker/qna-app/app/utils.py:340:23: F821 Undefined name `Pinecone`
ChatQnA/langchain/pinecone/ingest_intel.py:85:26: F821 Undefined name `doc_path`
ChatQnA/langchain/pinecone/rag_pinecone/chain.py:71:18: F821 Undefined name `TGI_LLM_ENDPOINT`
Found 5 errors.

File: ChatQnA/langchain/pinecone/rag_pinecone/__pycache__/__init__.cpython-311.pyc
Comment (lvliang-intel) [inline ChatQnA/langchain/pinecone/rag_pinecone/__pycache__/__init__.cpython-311.pyc:1]: Please remove the __pycache__ directory.

**Comment by pallavijaini0525** [comment]
I have made sure to pass all PRE-CI tests/pipeline tasks. anything is pending to get approval for this PR?

**Comment by lvliang-intel** [comment]
@pallavijaini0525，
Could you please move this PR to https://github.com/opea-project/GenAIComps/tree/main/comps/vectorstores/langchain
and https://github.com/opea-project/GenAIComps/tree/main/comps/dataprep?

**Comment by pallavijaini0525** [comment]
closing this PR and opening a new one for the same

**Pull request closed without merging**